### PR TITLE
docs: Add Databricks example

### PR DESCRIPTION
**Pull request by jrbourbeau**
This PR adds a new E2E example that walks through how to run Automodel on Databricks (single-GPU, multi-GPU, and multi-node). 

Closes https://github.com/NVIDIA-NeMo/Automodel/issues/826. Currently blocked by https://github.com/NVIDIA-NeMo/Automodel/issues/995. 

Some known TODOs:

- [ ] Update screenshots (right now it's just a single image being used as a placeholder)
- [ ] Remove temporary `--checkpoint.is_async True` workaround once https://github.com/NVIDIA-NeMo/Automodel/issues/1040 is resolved
- [ ] Double check installation init script before merging (e.g. if using async checkpointing, make sure pytorch 2.9+ is installed) 

cc @HuiyingLi @floraxhuang @adil-a for visibility

### CODE REVIEW

File: docs/guides/llm/databricks.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+# Model Training on Databricks
+
+Databricks is a widely-used platform for managing data, models, applications, and compute on the cloud. This guide shows how to use Automodel for scalable, performant model training on Databricks.
+
+The specific example here fine-tunes a [Llama-3.2-1B](https://huggingface.co/meta-llama/Llama-3.2-1B) model using the [SQuAD dataset](https://huggingface.co/datasets/rajpurkar/squad) from HuggingFace, but any Automodel functionality (for example, {doc}`model pre-training <pretraining>`, {doc}`VLMs </model-coverage/vlm>`, {doc}`other support models </model-coverage/overview>`) can also be run on Databricks.
```
Comment (jgerh) [inline docs/guides/llm/databricks.md]:
```suggestion
The specific example here fine-tunes a [Llama-3.2-1B](https://huggingface.co/meta-llama/Llama-3.2-1B) model using the [SQuAD dataset](https://huggingface.co/datasets/rajpurkar/squad) from Hugging Face, but any Automodel functionality (for example, {doc}`model pre-training <pretraining>`, {doc}`VLMs </model-coverage/vlm>`, {doc}`other support models </model-coverage/overview>`) can also be run on Databricks.
```

File: docs/guides/llm/databricks.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+## Compute
+
+Let’s start by [provisioning](https://docs.databricks.com/aws/en/compute/configure) a Databricks classic compute cluster with the following setup:
+
+- Databricks runtime: 17.3 LTS (Machine Learning version)  
+- Worker instance type: `g6e.12xlarge` on AWS (4x L40S GPU per node)  
+- Number of workers: 2  
+- Globally [environment variable](https://docs.databricks.com/aws/en/compute/configure#environment-variables): `GLOO_SOCKET_IFNAME=eth0` (see [this](https://docs.databricks.com/aws/en/machine-learning/train-model/distributed-training/spark-pytorch-distributor#gloo-failure-runtimeerror-connection-refused) for details)
```
Comment (jgerh) [inline docs/guides/llm/databricks.md]:
```suggestion
- Global [environment variable](https://docs.databricks.com/aws/en/compute/configure#environment-variables): `GLOO_SOCKET_IFNAME=eth0` (see [this](https://docs.databricks.com/aws/en/machine-learning/train-model/distributed-training/spark-pytorch-distributor#gloo-failure-runtimeerror-connection-refused) for details)   
```

File: docs/guides/llm/databricks.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+# Install Automodel + upgrade transformers to newer, compatible version
+/databricks/python3/bin/pip install --upgrade \
+    transformers \
+    git+https://github.com/NVIDIA-NeMo/Automodel
+```
+
+This will provision three compute nodes – one driver node we’ll attach a notebook to, and two worker nodes we’ll use for multi-node training.
+
+Note that we’ve selected a small number of instances for this example for demo purposes, but you can adjust the specific instance type and number of workers for your actual use case.
```
Comment (jgerh) [inline docs/guides/llm/databricks.md]:
```suggestion
Note that we’ve selected a small number of instances for demo purposes, but you can adjust the specific instance type and number of workers for your actual use case.
```

File: docs/guides/llm/databricks.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+This will provision three compute nodes – one driver node we’ll attach a notebook to, and two worker nodes we’ll use for multi-node training.
+
+Note that we’ve selected a small number of instances for this example for demo purposes, but you can adjust the specific instance type and number of workers for your actual use case. 
+
+## Training
+
+With the above compute resources provisioned, we’re ready to fine-tune a model using Automodel.
+
+Automodel uses YAML file recipes to configure various settings for the training process (for example, model, dataset, loss function, optimizer, etc.). Here we’ll use this [preconfigured recipe](https://github.com/NVIDIA-NeMo/Automodel/blob/main/examples/llm_finetune/llama3_2/llama3_2_1b_squad.yaml) for fine-tuning a Llama-3.2-1B model using the SQuAD dataset from HuggingFace.  In a notebook connected to our compute resource, download the training script and configuration file with these `curl` commands:
```
Comment (jgerh) [inline docs/guides/llm/databricks.md]:
```suggestion
Automodel uses YAML file recipes to configure various settings for the training process (for example, model, dataset, loss function, optimizer, etc.). Here we’ll use this [preconfigured recipe](https://github.com/NVIDIA-NeMo/Automodel/blob/main/examples/llm_finetune/llama3_2/llama3_2_1b_squad.yaml) for fine-tuning a Llama-3.2-1B model using the SQuAD dataset from Hugging Face. In a notebook connected to our compute resource, download the training script and configuration file with these `curl` commands:
```

File: docs/guides/llm/databricks.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+:::{figure} ./databricks-gpu-metrics.png
+:name: databricks-gpu-metrics-single
+:alt: Single GPU utilization of ~95% during model training.
+:align: center
+
+Single GPU utilization of ~95% during model training.
+:::
+
+To utilize all four GPUs available on this `g6e.12xlarge` instance use `torchrun --nproc-per-node=4` with our same training script and config file:
```
Comment (jgerh) [inline docs/guides/llm/databricks.md]:
```suggestion
To utilize all four GPUs available on this `g6e.12xlarge` instance, use `torchrun --nproc-per-node=4` with our same training script and config file: 
```

File: docs/guides/llm/databricks.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+Let’s start by [provisioning](https://docs.databricks.com/aws/en/compute/configure) a Databricks classic compute cluster with the following setup:
+
+- Databricks runtime: 17.3 LTS (Machine Learning version)  
+- Worker instance type: `g6e.12xlarge` on AWS (4x L40S GPU per node)  
+- Number of workers: 2  
+- Globally [environment variable](https://docs.databricks.com/aws/en/compute/configure#environment-variables): `GLOO_SOCKET_IFNAME=eth0` (see [this](https://docs.databricks.com/aws/en/machine-learning/train-model/distributed-training/spark-pytorch-distributor#gloo-failure-runtimeerror-connection-refused) for details)   
+- Cluster-scoped [init script](https://docs.databricks.com/aws/en/init-scripts/cluster-scoped):
+
+```
```
Comment (jgerh) [inline docs/guides/llm/databricks.md]: suggest the use of backticks with language identifier to render fenced code blocks (colorized text). Similar to ```py in line 117. 

```suggestion
```bash
```

File: docs/guides/llm/databricks.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+Note that we’ve selected a small number of instances for this example for demo purposes, but you can adjust the specific instance type and number of workers for your actual use case. 
+
+## Training
+
+With the above compute resources provisioned, we’re ready to fine-tune a model using Automodel.
+
+Automodel uses YAML file recipes to configure various settings for the training process (for example, model, dataset, loss function, optimizer, etc.). Here we’ll use this [preconfigured recipe](https://github.com/NVIDIA-NeMo/Automodel/blob/main/examples/llm_finetune/llama3_2/llama3_2_1b_squad.yaml) for fine-tuning a Llama-3.2-1B model using the SQuAD dataset from HuggingFace.  In a notebook connected to our compute resource, download the training script and configuration file with these `curl` commands:
+
+```
```
Comment (jgerh) [inline docs/guides/llm/databricks.md]:
```suggestion
```bash
```

File: docs/guides/llm/databricks.md [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+# Download training script
+!curl -O https://raw.githubusercontent.com/NVIDIA-NeMo/Automodel/refs/heads/main/examples/llm_finetune/finetune.py
+# Download configuration file
+!curl -O https://raw.githubusercontent.com/NVIDIA-NeMo/Automodel/refs/heads/main/examples/llm_finetune/llama3_2/llama3_2_1b_squad.yaml
+```
+
+Here’s what the model, dataset, and optimizer portions of the config file look like:
+
+```
```
Comment (jgerh) [inline docs/guides/llm/databricks.md]:
```suggestion
```yaml
```

File: docs/guides/llm/databricks.md [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+
+See the full file for complete details (`!cat llama3_2_1b_squad.yaml`). 
+
+### Single-node
+
+To run fine-tuning, we’ll use the `finetune.py` script from the Automodel repository and our config file.
+
+To run training on a single GPU, use this command:
+
+```
```
Comment (jgerh) [inline docs/guides/llm/databricks.md]:
```suggestion
```bash
```

File: docs/guides/llm/databricks.md [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+:name: databricks-gpu-metrics-single
+:alt: Single GPU utilization of ~95% during model training.
+:align: center
+
+Single GPU utilization of ~95% during model training.
+:::
+
+To utilize all four GPUs available on this `g6e.12xlarge` instance use `torchrun --nproc-per-node=4` with our same training script and config file: 
+
+```
```
Comment (jgerh) [inline docs/guides/llm/databricks.md]:
```suggestion
```bash
```

**Review by jgerh — COMMENTED** [review]
Completed tech pubs review and provided a few copyedits.
